public_html/wp-content/plugins/camptix/blocks/src/attendee-content/block.json

```diff
@@ -0,0 +1,12 @@
+{
+	"name": "camptix/attendee-content",
+	"icon": "welcome-view-site",
+	"category": "widgets",
+	"attributes": {},
+	"supports": {
+		"customClassName": false,
+		"className": false,
+		"html": false,
+		"multiple": false
+	}
+}
```

Comment:
I left those out since we didn't have access to `register_block_type_from_metadata` on WC.org yet, but now that we do I think I'll add that in 👍

Reply:
> adding a block setting to toggle whether accessing this particular piece of content automatically marks the person as attended or not. I can see a use case for this block of giving someone access to something before the event and not wanting it to actually mark them as attended.

+1

I think the option should be off-by-default too, since it doesn't seem like a safe assumption to me. That's especially true of in-person events.

Reply:
yep, that's what I went with in https://github.com/WordPress/wordcamp.org/pull/527/commits/d1eac20cd6e598126edcc8d957177f4457c63a49

Reply:
oh wait, what do you mean by "off-by-default"? I think it should default to marking people as attended, since in person events will most likely be the main users of this, for day-of-event content.

Reply:
It seems that could create a lot of false-positives or false-negatives, and I worry that the stats won't be reliable anymore.

* An organizer publishes content meant for viewing before the event. This one could probably be solved by only marking them as attended if the event is currently going on, though.
* An attendee who decides to stay home views the day-of content out of curiosity, but isn't actually at the event.
* Organizers use this block _instead of_ the check-in tool, but most attendees don't view the content because it's not essential and they're busy attending sessions, etc.

Reply:
> An organizer publishes content meant for viewing before the event. This one could probably be solved by only marking them as attended if the event is currently going on, though.

I'm back-and-forth on the idea of marking only when the event is happening… but maybe we should. An attendee viewing the content live & interacting with other attendees is probably getting more out of the event than someone basically watching youtube videos on their own later.

> An attendee who decides to stay home views the day-of content out of curiosity, but isn't actually at the event.

If there is online content and the attendee views it, don't you think that should be considered attendance? just because someone doesn't stick around doesn't mean we don't count them in (someone could also physically walk in, loop sponsors, and leave - they're still marked attended).

> Organizers use this block instead of the check-in tool

That's valid, but IMO unlikely. The check-in tool would still be the recommended way of checking in in-person, and works pretty seamlessly with a physical event's name badge pickup flow.

Reply:
Right now we're not tracking any attendance info from online events, so even if this isn't perfect, I think it's better than not collecting anything in.

Reply:
> don't you think that should be considered attendance?

No, that doesn't feel any different to me than watching the recordings on WPTV after an in-person event is over.

> someone could also physically walk in, loop sponsors, and leave - they're still marked attended.

That's true, but in practice I think the average amount of interaction with an in-person event is exponentially higher than online events. It's not a perfect measure, but nothing is.

I think the **interaction** is the key component for me, because that's the whole point of having the events IMO. Educational content is just the carrot, the interaction is what builds community.

Reply:
> The check-in tool would still be the recommended way

I'm not sure I see the point in having this option for in-person events then. Wouldn't they already be checked-in before they access the day-of content?

If so, then this is only relevant for online events, so I think the option should be always-off for in-person events.

> we're not tracking any attendance info from online events

That's a great point. I'm fine using "viewed day-of content" as a proxy for "attended an online event", just not in-person events.

Even for online events, though, I still think the button should be off by default. We have no way of knowing if the content inside the block is day-of content or something else.

Reply:
I understand what you're saying, but we're trying to build up the interaction piece for online events too. This isn't for hiding the livestream, so someone who only watches the talks wouldn't be "attended". This is for things like private zooms for hallway tracks, or other ways to interact with other attendees.

I imagine we'll see hybrid events before we move fully back in to in-person events, and even then I think someone who can't attend in-person (whether due to the current issues with covid, or more generally, a disability) but still gets involved should be considered attended.

I think "day of" content will be the majority use case. Even if not, [organizers can turn off the check-in feature.](https://github.com/WordPress/wordcamp.org/pull/527#issuecomment-652009735) And even if they don't, we'll only check-in people if they view it during the event. So I think for the least friction for most organizers, we should keep the default to on, "check in", for all events.

Reply:
Are you saying that it would only mark someone as attended if both of the following are true?

- it's an online event
- the person views the content while the event is happening

If so, then that's good enough for me.

Reply:
By default, we'll mark people attended only during the event, for any kind of event.

The organizer has the option to [turn this off per-block,](https://github.com/WordPress/wordcamp.org/pull/527#issuecomment-652009735) if they're adding content where that doesn't make sense.

Reply:
Yeah, I know there's an option to turn it off, but I don't think that's enough to keep the stats accurate. Most people will just go with whatever the default is, rather than paying close attention and considering the implications.

Although, that also means that if it were off by default, then there'd be lots of cases where the organizer should enable it, but doesn't, which would lead to inaccuracy in the other direction.

So maybe it'd make sense to set the default it to `on` for online events, but `off` for in-person events? That would reflect the correct setting in the majority of situations, and the organizer can still flip it if they want.